devlib/instrument/monsoon.py

```diff
@@ -0,0 +1,116 @@
+import csv
+import os
+import signal
+from subprocess import Popen, PIPE
+from tempfile import NamedTemporaryFile
+from devlib.instrument import Instrument, CONTINUOUS, MeasurementsCsv
+from devlib.exception import HostError
+from devlib.host import PACKAGE_BIN_DIRECTORY
+
+class MonsoonInstrument(Instrument):
+    """Instrument for Monsoon Solutions power monitor
+
+    To use this instrument, you need to install the monsoon.py script available
+    from the Android Open Source Project. As for May 2017 this is under the CTS
+    repository:
+
+    https://android.googlesource.com/platform/cts/+/master/tools/utils/monsoon.py
+
+    Collects power measurements only, from a selection of two channels, the USB
+    passthrough channel and the main output channel.
+
+    :param target: Ignored
+    :param monsoon_bin: Path to monsoon.py executable
+    :param device_entry: TTY device to use to communicate with the Power
+                         Monitor. If not provided, a sane default is used.
+    """
+    mode = CONTINUOUS
+
+    def __init__(self, target, monsoon_bin, device_entry=None):
```

Comment:
Used `device_entry` because that's what `energy_probe.py` uses but agree `tty_device` is better, will rename.